Util/File/read_until_match.m

```diff
@@ -0,0 +1,109 @@
+function [clines, number_of_lines] = read_until_match(fid, pattern, is_regex, preallocsize)
```

Comment:
function name doesn't match convention

Reply:
Yeah, the reason why was that these were all internal functions. However, I would like to avoid changing them now and have to rebase 3 other branches that are build on top of these commits.
Moreover, I will need to improve `test_docstring` anyway (support evaluation of docstring examples in classes), so these can go together later.

Reply:
or I can do a isolated commit on top of the last branch to make things easier.